app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordContents.java

```diff
@@ -0,0 +1,363 @@
+    }
+
+    // Helper method to get the list of locales and subtypes to display for this word
+    public ArrayList<LocaleRenderer> getLocalesList(final Activity activity) {
+
+        final SharedPreferences prefs = DeviceProtectedUtils.getSharedPreferences(activity.getApplicationContext());
+        final boolean localeSystemOnly = prefs.getBoolean(Settings.PREF_USE_SYSTEM_LOCALES, true);
+        final ArrayList<LocaleRenderer> localesList = new ArrayList<>();
+
+        // List of main language
+        final List<InputMethodSubtype> enabledMainSubtype = SubtypeSettingsKt.getEnabledSubtypes(prefs, true);
+        // List of system language
+        final List<Locale> enabledSystemLocale = SubtypeSettingsKt.getSystemLocales();
+        // List of user dictionary
+        final TreeSet<String> userDictionaryList = UserDictionaryListFragment.getUserDictionaryLocalesSet(activity);
+        // To combine lists
+        final TreeSet<String> languageList = new TreeSet<>(String::compareToIgnoreCase);
+
+        // Add languages from the user dictionary list
+        // (local system languages and dictionary languages in which a word is already saved)
+        if (userDictionaryList != null) {
+            languageList.addAll(userDictionaryList);
+        }
+
+        // mLocale is removed from the language list as it will be added to the top of the list
+        languageList.remove(mLocale);
+
+        // Add the main language selected in the "Language and Layouts" setting except "No language"
+        for (InputMethodSubtype mainSubtype : enabledMainSubtype) {
+            Locale mainLocale = LocaleUtils.constructLocaleFromString(mainSubtype.getLocale());
+            String mainLocaleString = mainLocale.toString();
+            if (userDictionaryList != null && !userDictionaryList.contains(mainLocale.toString())
+                    && !mainLocaleString.equals(mLocale) && !mainLocaleString.equals("zz")) {
+                languageList.add(mainLocaleString);
+            }
+            // Secondary language is added only if main language is selected and if system language is not enabled
+            if (!localeSystemOnly) {
+                for (Locale secondSubtype : Settings.getSecondaryLocales(prefs, mainSubtype.getLocale())) {
+                    String secondLocaleString = secondSubtype.toString();
+                    // Add secondary subtypes if they are not included in the user's dictionary
+                    if (userDictionaryList != null && !userDictionaryList.contains(secondSubtype.toString()) && !secondLocaleString.equals(mLocale)) {
+                        languageList.add(secondLocaleString);
+                    }
+                }
+            }
+        }
+
+        for (Locale systemSubtype : enabledSystemLocale) {
+            String systemLocaleString = systemSubtype.toString();
+            if (userDictionaryList != null && !userDictionaryList.contains(systemSubtype.toString()) && !systemLocaleString.equals(mLocale)) {
```

Comment:
No need to do `systemSubtype.toString()` when you already have `systemLocaleString`

Reply:
Change made with commit 0582df5